### PR TITLE
feature: impl From for CfKV

### PR DESCRIPTION
**Pull request by wenbobuaa**
# Description

给 CfKV 实现了 from

## Type of change

- **New feature**     

# Checklist:

- [x] **Style**:       My code follows the **style guidelines** of this project
- [x] **Self-review**: I have performed a **self-review** of my own code
- [ ] **Comment**:     I have **commented my code**, particularly in hard-to-understand areas
- [ ] **Doc**:         I have made corresponding changes to the **documentation**
- [x] **No-warnings**: My changes generate **no new warnings**
- [x] **Add-test**:    I have added **tests** that prove my fix is effective or that my feature works
- [x] **Pass**:        New and existing **unit tests pass** locally with my changes
- [ ] **Dep**:         Any **dependent** changes have been merged and published in downstream modules

### CODE REVIEW

File: components/epaxos/src/snapshot/rocksEngine/engine.rs [hunk 1]
```diff
@@ -3,12 +3,23 @@ use super::{DBColumnFamily, Error, RocksDBEngine};
...
     cf: &'a DBColumnFamily,
     k: &'a [u8],
     v: &'a [u8],
 }
 
+impl<'a> From<(&'a DBColumnFamily, &'a [u8], &'a [u8])> for CfKV<'a> {
+    fn from(cfkv: (&'a DBColumnFamily, &'a [u8], &'a [u8])) -> CfKV<'a> {
+        CfKV {
+            cf: cfkv.0,
+            k: cfkv.1,
```
Comment (drmingdrmer) [inline components/epaxos/src/snapshot/rocksEngine/engine.rs]: From主要是为了简化琐碎的测试这种代码, 按这个目的话, 在给个`From<(str,str,str)>`的实现更方便,
然后就可以`x.mset(("default", "foo", "bar").into())`了

File: components/epaxos/src/snapshot/enum_str.rs [hunk 2]
```diff
@@ -15,12 +16,23 @@ macro_rules! enum_str {
                     ),*
                 }
             }
+
+            fn from_str<'a> (val: &str) -> Result<&'a Self, String> {
```
Comment (drmingdrmer) [inline components/epaxos/src/snapshot/enum_str.rs]: 感觉这样的话直接实现`From<&str>` 更规范一点.

使用起来好像也挺方便:`xx.from("foo")` 和`"foo".into()`. 有上下文的地方大部分都可以类型推断不需要写`DBColumnFamily`.

Comment (wenbobuaa) [reply components/epaxos/src/snapshot/enum_str.rs]: 实现 `From` 的话，返回值不能是 `Result`。但是从 `str` 到 `DBColumnFamily` 有可能出现不匹配，要报 `Error` 。所以才写了个 `from_str`。

Comment (drmingdrmer) [reply components/epaxos/src/snapshot/enum_str.rs]: 深谋远虑!

**Pull request merged**